src/utils/RegistryTrait.php

```diff
@@ -90,9 +96,17 @@ private static function _registryFromString(string $name) : object{
 		if(!isset(self::$members[$upperName])){
 			throw new \InvalidArgumentException("No such registry member: " . self::class . "::" . $upperName);
 		}
-		return self::preprocessMember(self::$members[$upperName]);
+		/** @phpstan-var T $return */
+		$return = self::preprocessMember(self::$members[$upperName]);
+		// We need to force typing here, otherwise PHPStan understands that it's an object and not T
```

Comment:
This seems a bit suspicious. I really don't like using @var when the type should be obvious.